### PR TITLE
comments: Add CommentName to AstInfo

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR cleans up some rough edges that exist for comments around name resolution of objects. Previously we relied on types like `UnresolvedItemName` that we'd then "manually" resolve in planning. In this PR we refactor the `COMMENT ON` code paths to use the associated types on the `AstInfo` trait, e.g. `AstInfo::ItemName`, which performs automatic name resolution.

This PR also adds a new associated type to `AstInfo` called `ColumnName` which provides a similar level of automatic name resolution as the other associated types.

### Motivation

   * This PR refactors existing code.

### Tips for reviewer

The PR is broken into two commits:

1. Refactors the existing variants of `CommentObjectType` to use `T::ItemName` instead of `UnresolvedItemName`.
2. Adds a new associated type to `AstInfo` called `CommentName`

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

**Comment by moulimukherjee** [comment]
Thanks for doing this! I will rebase https://github.com/MaterializeInc/materialize/pull/22019 over this PR, will simplify the column validation bits.

File: src/sql/src/names.rs [hunk 1]
```diff
@@ -519,6 +520,35 @@ impl std::fmt::Display for ResolvedItemName {
     }
 }
 
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ResolvedColumnName {
```
Comment (moulimukherjee) [inline src/sql/src/names.rs:524]: OOC should this be added to `ResolvedObjectName` here https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/names.rs#L780 as well?

Comment (moulimukherjee) [reply src/sql/src/names.rs:524]: nvm ^, since it's not an object and will not have a corresponding `ObjectId`

Comment (ParkMyCar) [reply src/sql/src/names.rs:524]: Good thought! I noodled on this for a bit and we could maybe add this to `ResolvedObjectName` and return the relation's ID, there didn't seem to be a need to do this at the moment, so I held off

**Review by mjibson — APPROVED** [review]

**Pull request merged**